how-to-ns/src/com/gfredericks/how_to_ns/main.clj

```diff
@@ -54,44 +54,55 @@
         (println "Exception in how-to-ns when checking" (str file))
         (prn e)))))
 
+(def default-paths
+  ["src" "test"])
+
 (defn check
-  [paths opts]
-  (->> (all-clojure-files paths)
-       (map (fn [file]
-              (try
-                (let [contents (slurp file)
-                      formatted (how-to-ns/format-initial-ns-str contents opts)]
-                  (if (= contents formatted)
-                    0
-                    (do
-                      (binding [*out* *err*]
-                        (println "Bad ns format:")
-                        (println (unified-diff
-                                  (str file)
-                                  contents
-                                  formatted)))
-                      1)))
-                (catch Exception e
-                  (report-file-specific-exception file e)
-                  1))))
-       (reduce +)))
+  ([options]
+   (let [paths (or (:paths options) default-paths)
+         opts (merge how-to-ns/default-opts options)]
```

Comment:
is there a reason to do this merge here? surely it's already being done somewhere else, since we didn't have it before?

Reply:
It was in lein plugin and i can't reuse it from here. Also logic slightly changed, because of how Clj tools works